lib/utils/formatting.ts

```diff
@@ -1,15 +1,50 @@
+import {
+  camelCase,
+  kebabCase,
+  pascalCase,
+  snakeCase
+} from 'case-anything';
+
+export type CaseType =
+    | 'camel'
+    | 'kebab'
+    | 'snake'
+    | 'pascal'
+    | 'kebab-or-snake';
+
 /**
  *
  * @param str
- * @returns formated string
- * @description normalizes input to supported path and file name format.
- * Changes camelCase strings to kebab-case, replaces spaces with dash and keeps underscores.
+ * @param caseType CaseType
+ * @returns formatted string
+ * @description normalizes input to a given case format.
+ * Options are: "camel" | "kebab" | "snake" | "pascal" | "kebab-or-snake"
  */
-export function normalizeToKebabOrSnakeCase(str: string) {
-  const STRING_DASHERIZE_REGEXP = /\s/g;
-  const STRING_DECAMELIZE_REGEXP = /([a-z\d])([A-Z])/g;
-  return str
-    .replace(STRING_DECAMELIZE_REGEXP, '$1-$2')
-    .toLowerCase()
-    .replace(STRING_DASHERIZE_REGEXP, '-');
-}
+export const normalizeToCase = (
+  str: string,
+  caseType: CaseType = 'kebab',
+) => {
+  switch (caseType) {
+    case 'camel':
+      return camelCase(str);
+    case 'kebab':
+      return kebabCase(str);
+    case 'pascal':
+      return pascalCase(str);
+    case 'snake':
+      return snakeCase(str);
+    case 'kebab-or-snake':
+      return kebabCase(str, { keep: ['_', '@', '/', '.'] })
+    default:
+      throw new Error(`Case type ${caseType} is not supported.`);
+  }
+};
+
+export const formatString = (str: string) => {
```

Comment:
can you elaborate here in the code on what this function is supposed to do?

github copilot can help :p

Reply:
Dang, I'm surprised copilot understood immediately the code. I'm starting to question reality :D 

Added some JsDoc. Basically this function escapes parenthesis which will then later be removed. 

I see here an opportunity to create `@nestjs/utils` or `@nestjs/stringUtils` because this code is shared across projects, and I feel it would benefit to centralize it so that it's easier to maintain and can be reused by plugin developers. I didn't do it to keep the scope limited, but it shouldn't be too hard to do.

Reply:
ok I updated the same code also in `@nestjs/schematics`